tutorials/surrogate/cifar10_example.py

```diff
@@ -0,0 +1,429 @@
+        # self.log("val_loss", loss_val)
+        # self.log("val_acc", val_acc_val)
+        return loss_val
+
+    def configure_optimizers(self) -> torch.optim.Optimizer:
+        return torch.optim.Adam(self.parameters(), lr=self.lr, weight_decay=self.weight_decay)
+
+
+def get_data_loaders(batch_size: int) -> Tuple[DataLoader, DataLoader]:
+    """
+    Get CIFAR10 train and validation dataloaders.
+
+    Parameters
+    ----------
+    batch_size: int
+                batch size
+
+    Returns
+    -------
+    DataLoader
+        training dataloader
+    DataLoader
+        validation dataloader
+    """
+    data_transform = Compose([ToTensor(),
+                              Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
+
+    # set empty DataLoader
+    train_loader = DataLoader(
+            dataset=TensorDataset(torch.empty(0), torch.empty(0)),
+            batch_size=batch_size,
+            shuffle=False)
+
+    if MPI.COMM_WORLD.Get_rank() == 0:  # Only root downloads data.
+        train_loader = DataLoader(
+            dataset=CIFAR10(
+                download=True, root=".", transform=data_transform, train=True
+            ),  # Use CIFAR10 training dataset.
+            batch_size=batch_size,  # Batch size
+            shuffle=True,  # Shuffle data.
+        )
+
+    if not hasattr(get_data_loaders, "barrier_called"):
+        MPI.COMM_WORLD.Barrier()
+
+        setattr(get_data_loaders, "barrier_called", True)
+
+    if MPI.COMM_WORLD.Get_rank() != 0:
+        train_loader = DataLoader(
+            dataset=CIFAR10(
+                download=False, root=".", transform=data_transform, train=True
+            ),  # Use CIFAR10 training dataset.
+            batch_size=batch_size,  # Batch size
+            shuffle=True,  # Shuffle data.
+        )
+    val_loader = DataLoader(
+        dataset=CIFAR10(
+            download=False, root=".", transform=data_transform, train=False
+        ),  # Use CIFAR testing dataset.
+        batch_size=1,  # Batch size
+        shuffle=False,  # Do not shuffle data.
+    )
+    return train_loader, val_loader
+
+
+def ind_loss(
+        params: Dict[str, Union[int, float, str]],
+) -> Generator[float, None, None]:
+    """
+    Loss function for evolutionary optimization with Propulate. Minimize the model's negative validation accuracy.
+
+    Parameters
+    ----------
+    params: dict[str, int | float | str]]
+
+    Returns
+    -------
+    Generator[float, None, None]
+        yields the negative validation accuracy of the trained model
+    """
+    # Extract hyperparameter combination to test from input dictionary.
+    lr = float(params["lr"])  # Learning rate
+    perm = str(params["perm"])  # Permutation
+
+    epochs: int = 2  # Number of epochs to train
+
+    rank: int = MPI.COMM_WORLD.Get_rank()  # Get rank of current worker
+
+    num_gpus = torch.cuda.device_count()  # Number of GPUs available
+    if num_gpus == 0:
+        device = torch.device('cpu')
```

Comment:
```suggestion
        device = torch.device("cpu")
```
If you use `black` with your code, it should automatically convert single to double quotes etc.